### PR TITLE
Add a flag to graphman to list all deployments

### LINKED ISSUE #3407: graphman: provide an official way to list all deployments
`graphman` should have a way to list all deployments. Currently, and solely as a sideeffect of how it handles its arguments, `graphman info ''` will do that, but there should be a better way for that (maybe `graphman info -l` or `-a`)

### PR DESCRIPTION
**Pull request by incrypto32**
Fixes #3407

### CODE REVIEW

File: node/src/bin/manager.rs [hunk 1]
```diff
@@ -966,14 +969,32 @@ async fn main() -> anyhow::Result<()> {
...
+                None => {
+                    if all {
+                        let deployment = DeploymentSearch::All;
+                        commands::info::run(primary, store, deployment, current, pending, used)
+                            .err();
+                    } else {
+                        eprintln!(
+                            "Please specify a deployment hash or use --all to list all deployments"
+                        );
+                        std::process::exit(1);
```
Comment (lutter) [inline node/src/bin/manager.rs:994]: You can just `bail!("Please specify a deployment or use --all to list all deployments)` here (using `anyhow::bail!`) - that's what all other errors end up doing and there's no need to explicitly `exit(1)`

File: node/src/manager/deployment.rs [hunk 2]
```diff
@@ -42,6 +43,7 @@ impl fmt::Display for DeploymentSearch {
                 hash,
                 shard: Some(shard),
             } => write!(f, "{}:{}", hash, shard),
+            DeploymentSearch::All => write!(f, ""),
```
Comment (lutter) [inline node/src/manager/deployment.rs:46]: you could just do `{ Ok(()) }` here and not call `write!` at all

**Review by lutter — CHANGES_REQUESTED** [review]
Looks great! Just change how the error is returned using `bail!` and we can merge this

**Comment by incrypto32** [comment]
@lutter Im not sure why the integration tests fail, let me know if i can mark this PR ready for review.

**Review by lutter — APPROVED** [review]
This looks great, and is ready to be merged (pending some unrelated fixes to our CI)

Before merging, can you squash all the commits into one and use this commit message:
```
node: Add a flag to graphman to list all deployments 

Fixes https://github.com/graphprotocol/graph-node/issues/3407
```

**Comment by lutter** [comment]
The integration tests are fixed now; just rebase your changes on top of the latest master, and all checks should pass

**Comment by incrypto32** [comment]
@lutter  looks like all tests passed!

**Pull request merged**